test/test_unistd.rs

```diff
@@ -9,16 +9,16 @@ use std::ffi::CString;
 use std::fs::File;
 use std::io::Write;
 use std::os::unix::prelude::*;
-use std::env::current_dir;
 use tempfile::tempfile;
 use tempdir::TempDir;
-use libc::off_t;
+use libc::{_exit, off_t};
 
 #[test]
 fn test_fork_and_waitpid() {
+    let m = ::FORK_MTX.lock().expect("Mutex got poisoned by another test");
     let pid = fork();
     match pid {
-        Ok(Child) => {} // ignore child here
+        Ok(Child) => { unsafe { _exit(0) }; }
```

Comment:
I initially used `_exit(0)` because I didn't know exactly where the deadlock was, and `_exit(0)` skips virtually all teardown.  But I just tried it, and it looks like `std::process::exit(0)` works too.  I guess the deadlock is probably in the test harness code somewhere.  I'll switch.